includes/Modules/Analytics_4.php

```diff
@@ -1173,39 +1174,36 @@ function ( $metric_def ) {
 	 * Parses the orderby value of the data request into an array of AnalyticsData OrderBy object instances.
 	 *
 	 * @since 1.93.0
+	 * @since n.e.x.t Updated to provide support for ordering by dimensions.
 	 *
 	 * @param array|null $orderby Data request orderby value.
 	 * @return Google_Service_AnalyticsData_OrderBy[] An array of AnalyticsData OrderBy objects.
 	 */
 	protected function parse_reporting_orderby( $orderby ) {
-		if ( empty( $orderby ) || ! is_array( $orderby ) ) {
+		if ( empty( $orderby ) || ! is_array( $orderby ) || ! wp_is_numeric_array( $orderby ) ) {
 			return array();
 		}
 
 		$results = array_map(
 			function ( $order_def ) {
-				$order_def = array_merge(
-					array(
-						'fieldName' => '',
-						'sortOrder' => '',
-					),
-					(array) $order_def
-				);
-
-				if ( empty( $order_def['fieldName'] ) || empty( $order_def['sortOrder'] ) ) {
+				$order_by = new Google_Service_AnalyticsData_OrderBy();
+				$order_by->setDesc( ! empty( $order_def['desc'] ) );
+
+				if ( isset( $order_def['metric'] ) && isset( $order_def['metric']['metricName'] ) ) {
+					$metric_order_by = new Google_Service_AnalyticsData_MetricOrderBy();
+					$metric_order_by->setMetricName( $order_def['metric']['metricName'] );
+					$order_by->setMetric( $metric_order_by );
+				} elseif ( isset( $order_def['dimension'] ) && isset( $order_def['dimension']['dimensionName'] ) ) {
+					$dimension_order_by = new Google_Service_AnalyticsData_DimensionOrderBy();
+					$dimension_order_by->setDimensionName( $order_def['dimension']['dimensionName'] );
+					$order_by->setDimension( $dimension_order_by );
+				} else {
```

Comment:
Currently we don't support neither `metric` nor `dimension` properties in the order by object. The only available property with a field name is `fieldName`. I think we can consider it to be a dimension name because as far as I remember we sort everything using the `date` dimension only (i might be wrong, don't remember exactly). Let's make it compatible with the current approach and leave a comment and maybe create a separate ticket to allow `metric`/`dimension` properties in the sort object for Analytics-4 reports.